psalm.xml

```diff
@@ -42,6 +42,40 @@
                 <file name="lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php"/>
             </errorLevel>
         </FalsableReturnStatement>
+        <ForbiddenCode>
+            <errorLevel type="suppress">
+                <!-- See https://github.com/doctrine/dbal/pull/3934 -->
+                <file name="lib/Doctrine/DBAL/Logging/EchoSQLLogger.php"/>
+
+                <!-- The call to var_dump() here is by design -->
```

Comment:
If this is by design, shouldn't it be handled in the file itself in the affected line of code to provide this context/decision?

Reply:
Do you suggest to use in-file annotations instead of suppressing the error here? I'd like to avoid having annotations unrelated to the code if possible. Additionally, this entire file is a dumper, so no matter where in the file `var_dump()` is called, it's by design.

Reply:
> Additionally, this entire file is a dumper, so no matter where in the file var_dump() is called, it's by design.

I guess this answer counts for letting it stay in psalm.xml.
A lot of the suppressed rules are supposed to be fixed at one time or another, but I tend to use inline annotation when the affected code looks like this by design (or can't be disabled more selective in the config with the analysis tool of choice).